text/0000-payment-proofs.md

```diff
@@ -0,0 +1,101 @@
+* `receiver_address` - An ed25519 public key for the receiver, typically the public key of the user's v3 onion address.
+* `receiver_signature` - A signature of the sender_address, received amount, and kernel commitment that validates against the `receiver_address`. 
+
+## Generating proofs
+
+Receipt confirmations (`receiver_signature`s) will be generated by the payee by providing an ED25519 signature of: `(amount || kernel_commitment || sender_address || message)`, using the private key of the `receiver_address`.
+
+If a `sender_address` was provided, then `sender_signature` can be generated for `(amount || kernel_commitment || sender_address || message)` using the private key of the `sender_address`.
+
+Sender will then create and store the following info, which can be considered the complete `payment_proof`:
+
+* `receiver_address`
+* `receiver_signature`
+* `amount`
+* `kernel_commitment`
+* `sender_address` (OPTIONAL)
+* `sender_signature` (OPTIONAL)
+* `message` (OPTIONAL)
+
+## Verifying Proofs
+
+This `payment_proof` can be provided by the sender at any time to convince a payee that a payment was made to them. The proof can be verified as follows:
+
+1. Ensure the `kernel_commitment` is confirmed on-chain.
+2. Verify that the `receiver_address` belongs to the payee.
+3. Verify that the `receiver_signature` is valid.
+4. If a `sender_address` was included, verify that a valid `sender_signature` was provided.
+
+## Wallet actions
+### init-send
+
+As part of the first step of the tx-building process, the sender/payer can optionally choose to generate the `sender_address` using their keychain. 
+
+The `receiver_address`, optional keychain path of the `sender_address`, and the optional `message` must be stored locally, along with the `slate_id`.
+
+The `sender_address`, `receiver_address`, and `messsage` will then be added to the `payment_proof` structure of the slate.
+
+### receive
+
+If the `payment_proof` structure exists on the slate, it is mandatory that the `receiver_signature` is generated and added to the slate as part of the receive tx-building step.
+
+### finalize
+
+Using the `slate_id`, the sender can retrieve the original `sender_address`, `receiver_address`, and `message` that were included in the slate, and verify that those fields remain unchanged. The sender must then validate the `receiver_signature`. If any of the original `payment_proof` slate fields were modified, or if the `receiver_signature` is invalid, the transaction must be rejected by the sender.
+
+Once the `payment_info` details have been validated, the sender can generate and store the `payment_proof` (See `Generating Proofs` above), and then finalize the transaction as normal.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+None
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+* Should the `sender_address` and `sender_signature` be optional, required, or removed entirely?
+* Can this be adapted to work for invoices?
```

Comment:
Can this RFC incorporate such scheme for invoices or is the logic/flow incompatible here?

Reply:
I don't think invoices will work under this scheme as defined:

* Payee generates payment request invoice.. they can add the `receiver_address` and `amount` at this point, but obviously not enough to generate the signature as there's no kernel
* Invoice sent to payer, all they can do at this point is optionally add the `sender_signature` and can't validate the Payee's proof.
* Invoice is sent back to payer, who can complete the `recipient_signature` and return it if they like at this point, but can also simply finalize and post while ignoring any request to fill out a proof.

@lehnberg does grinbox handle payment proofs using invoices?

Reply:
I believe proposed scheme cannot, David and I discussed it in Keybase IIRC. 

> does grinbox handle payment proofs using invoices?

@yeastplume I believe it doesn't as [the grinbox scheme](https://github.com/vault713/wallet713/blob/master/docs/usage.md#transaction-proofs-grinbox-only) would face the same problem, but actually not 100% sure. @jaspervdm?

Reply:
Correct, invoice payment proofs arent supported on grinbox either.